ros2 urg_node parameter file

installed sudo apt install ros-foxy-urg-node.
ros2 run urg_node getID works fine and returns ID.
ros2 run urg_node urg_node_driver __params:=/opt/ros/foxy/share/urg_node/launch/urg_node_ethernet.yaml

returns can't find serial port 115200 /dev/cu.usbmodemxxxxx

I've set the ip address in the yaml file and it just doesent seem to be reading the paramtere file anywere.  Any idea on why its not readin the parameter file and using the IP address of the hokuyo (since thats the model I have) ?

Tried to git clone and build, but colcon build comes out with a bunch of catkin (ros1) type of errors.  

Thanks for any assistance.  Apologies if this is an idiot request.  I've read everything I could find and search upon.  Thanks


urg_node_ethernet.yaml file :
```
urg_node:
  ros__parameters:
    angle_max: 3.14
    angle_min: -3.14
    ip_address: "192.168.1.68"
    ip_port: 10940
    serial_baud: 57600
    laser_frame_id: "laser"
    calibrate_time: false
    default_user_latency: 0.0
    diagnostics_tolerance: 0.05
    diagnostics_window_time: 5.0
    error_limit: 4
    get_detailed_status: false
    publish_intensity: false
    publish_multiecho: false
    cluster: 1
    skip: 0
```

Updated the original file.  the orignal is run with urg_node being installed as a binary (apt install).  I realized that I was pulling from the wrong branch last night as it dawned on me .. . lol.  I'll try building it from source, but was wondering why the binary one wasnt able to take the param file and seemingly was ignoring it. 

I've built the ros2 version (ros2-devel).  sourceed it (. install/local_setup.bash).  and then executed the following with the same yaml file (only thing changed from the default is the ip address definition).

ros2 run urg_node urg_node_driver __params:=/home/edf/ros2_ws/src/urg_node/launch/urg_node_ethernet.yaml
[ERROR] [1604762567.286880359] [urg_node]: Error connecting to Hokuyo: Could not open serial Hokuyo:
/dev/cu.usbmodem141101 @ 115200
could not open serial device.
[ERROR] [1604762567.287887545] [urg_node]: Error connecting to Hokuyo: Could not open serial Hokuyo:
/dev/cu.usbmodem141101 @ 115200


Obviously I can ping the Ip of the scanner.  But it doesnt even seem to be going to the param file to pick up the IP network device and seems to be just defaulting to the usb device.  

As another test, I changed the baud rate on all the yaml devices to 57600 just to see if it was taking the baud rate from the yaml files.  And it still gives the same error above - so I think I must be passing the wrong parameter in for the yaml file as it just isnt taking any of them. 

and finally, I populated the ip address (hard coded) in the urg_node.cpp file  at the UrgNode section, rebuilt it and it works fine.  ros2 topic echo /scan shows the data from the scanner.  So the build obviously works, just that I cant seem to feed a parameter file in correctly.  its probably an obvious thing I'm missing, I just dont know what it is. . lol

--- a/README.md
+++ b/README.md
@@ -16,7 +16,7 @@ Until the launch API is sorted out, there are two ways to view the laserscan in 
 1) use the static_transform_publisher tool from tf2_ros to publish a static transform for a fixed frame simultaneously with rviz and urg_node.
 
 ```
-ros2 run tf2_ros static_transform_publisher 0 0 0 0 0 0 "world" "laser"
+ros2 run tf2_ros static_transform_publisher --frame-id world --child-frame-id laser
 ```
 
 where "laser" is the frame id, which you can set in your urg_node yaml file
@@ -32,6 +32,15 @@ A urdf file is already included and if you have succesfully ran an `colcon build
 ```
 <path to ros2_ws>/install/share/urg_node/launch/hokuyo_laser.urdf
 ```
+
+Now run RViz:
+
+```
+ros2 run rviz2 rviz2
+```
+or simply `rviz2`
+
+Add the LaserScan topic and change the global fixed frame from 'map' to 'world' to display the laser scan.
 
 
 ### Parameters
@@ -85,15 +94,15 @@ For example to reconfigure the cluster parameter using command line :
 ros2 param set /urg_node cluster 1
 ```
 
-### How to use the ust-20lx (and other ethernet based laser)
+### How to use the ust-20lx (and other Ethernet-based laser)
 
-To use ust-20lx, you need to be on the same subnet as the laser.
-The ust-20lx default ip is 192.168.0.10, so you might need to change your ip, for something on the same subnet.
+To use ust-20lx you need to be on the same subnet as the laser.
+The ust-20lx default IP address is 192.168.0.10, so you might need to change your IP to something on the same subnet.
 
 On Ubuntu :
-- go to the network settings
-- settings of the wired connection
-- under IPv4, change the IPv4 method from automatic to Manual and under Addresses, set the values:
+- Network settings
+- Wired connection
+- Under IPv4, change the IPv4 method from automatic to Manual and under Addresses, set the values:
 
 ```
     Address : 192.168.0.15
@@ -101,12 +110,13 @@ On Ubuntu :
     Gateway : 192.168.0.1
 ```
 
-- Turn off and on the connection or wait a bit for the change to apply
+- Turn the connection off and on again to apply the change
 - To check if it worked, open a terminal and type
 
 ```
-ifconfig
+ip --human-readable --color address
 ```
+or simply `ip a`
 
 - Under eth0 (or maybe something like enpxxxxx), your IP should be 192.168.0.15.
 You should now be able to ping the ust-20lx at its address (by default 192.168.0.10)
@@ -115,24 +125,10 @@ You should now be able to ping the ust-20lx at its address (by default 192.168.0
 ping 192.168.0.10
 ```
 
-If you don't receive any answer, you might have a connection problem or the IP of your laser might have been change, either find it and go on the same subnet or reset it.
+If you don't receive any answer, you might have a connection problem or the IP of your laser might have changed. If the IP was changed, you can either change your computer's IP to the same subnet or reset the laser to factory settings.
 
-- Once you can ping the laser, you can launch the urg_node_driver:
+- Once you can ping the laser, launch the urg_node_driver:
 
 ```
 ros2 run urg_node urg_node_driver --ros-args --params-file path/to/my/file.yaml
 ```
-
-then the static publisher:
-
-```
-ros2 run tf2_ros static_transform_publisher 0 0 0 0 0 0 "world" "laser"
-```
-
-and rviz:
-
-```
-ros2 run rviz2 rviz2
-```
-
-Add the laserScan topic, change the frame on rviz from map to world and you should be able to see the laser.

--- a/README.md
+++ b/README.md
@@ -114,9 +114,9 @@ On Ubuntu :
 - To check if it worked, open a terminal and type
 
 ```
-ip --human-readable --color address
+ip --color address
 ```
-or simply `ip a`
+or simply `ip -c a`
 
 - Under eth0 (or maybe something like enpxxxxx), your IP should be 192.168.0.15.
 You should now be able to ping the ust-20lx at its address (by default 192.168.0.10)

--- a/README.md
+++ b/README.md
@@ -64,7 +64,7 @@ urg_node:
 To give parameters to urg_node :
 
 ```
-ros2 run urg_node urg_node_driver __params:=path/to/my/file.yaml
+ros2 run urg_node urg_node_driver --ros-args --params-file path/to/my/file.yaml
 ```
 
 You can reconfigure parameters while the node is launched.
@@ -120,7 +120,7 @@ If you don't receive any answer, you might have a connection problem or the IP o
 - Once you can ping the laser, you can launch the urg_node_driver:
 
 ```
-ros2 run urg_node urg_node_driver __params:=path/to/my/file.yaml
+ros2 run urg_node urg_node_driver --ros-args --params-file path/to/my/file.yaml
 ```
 
 then the static publisher:

